chapters/05-version-control-solo.Rmd

```diff
@@ -12,6 +12,115 @@ Clearly identify which tasks are best accomplished by each interface.
+* apply the change-stage-commit process as the Git workflow for tracking changes and viewing a repository's commit history
+* create branches to isolate work in progress and protect the main branch while working on major changes
+* publish repositories to GitHub and coordinate between remote and local versions
+
+## Overview
+
+All text (no code):
+* why to use version control
+* why to use Git as a version control system (popularity, GitHub, 
+* how are Git and GitHub different?
+  * local vs remote
+  * publishing 
+* why use version control for software engineering 
+  * basics of version control and why it matters for code
+  * preview this chapter, connect to other chapters
+
+## Git integration in RStudio
+
+### Setting up Git in RStudio
+
+* text: connecting Git and RStudio
+  * accessing Git through Global options
+  * tab in environment pane 
+* code: `usethis::use_git` to initialize repository
+* text: explain what happens when repo is initialized
+  * creates hidden folder containing git history
+  * some files ignored (note that it's possible to add other files to ignore)
+
+### Tracking changes in RStudio
+
+All text and images:
+* exercise: work through basic workflow using RStudio integration
+  * all existing files in repo appear in git pane
+  * ensure box is checked so these files are staged
+  * click Commit button, add commit message and click commit
+  * note pop up box with commit report, files no longer appear in window
+* reiterate workflow by editing README
+  * status difference between tracking new files and editing previously tracked file
+
+* exercise: create new file, commit, delete file, commit
+
+### Viewing history in RStudio
+
+All text and images:
+* click on clock button in Git pane (don't confuse with history tab; all git buttons are in Git tab)
+* explain fields: subject, author, date, SHA
+* click on a previous commit to see changes
+
+## Git in Terminal
+
+* text: git on command line includes commands for full range of tasks, including those we've already covered:
+  * initialize repository: `git init`
+  * basic workflow: `git add` then `git commit`
+  * view history: `git log`
+
+### Exploring history in Terminal
+
+* code: make a change to README, then restore to previous (most recent) commit (pretend changes didn't exist, as if we changed our mind)
+  * `git status`
+  * `git restore README.md`
+  * `git status`
+
+* text: if we wanted to use content from a previous commit, like the file we created then deleted earlier
+  * find SHA key in log
+* code:
+  * `git checkout XXXXXXX filename`
+  * `git status`
+  * `git restore --staged filename`
+  * `git restore filename`
+  * `git status`
+
+### Creating branches in Terminal
+
+* text: what is a branch? 
+  * main branch (make sure usethis creates main as default branch)
+  * why work with them?
+* code: create branch, make changes, merge with main
+  * `git branch`
```

Comment:
This is a better idea for using the Terminal. Though, personally, if we could stay in R and use e.g. the gert package to do these things, that would be more ideal (and easier to automate building the book and example packages within the book). gert is a really nicely developed package by the people at rOpenSci, so the documentation is quite good. Plus discoverability of Git functions is much easier than with the Terminal.

For instance for creating and checking out a branch, it is: `gert::git_branch_checkout()`.